e2e/_suites/ingest-manager/features/fleet_mode_agent.feature

```diff
@@ -40,5 +36,33 @@ Scenario: Re-enrolling an agent
 @revoke-token
 Scenario: Revoking the enrollment token for an agent
   Given an agent is deployed to Fleet
+    And the agent is un-enrolled
+    And the "agent" process is "stopped" on the host
   When the enrollment token is revoked
-  Then an attempt to enroll a new agent fails
+  Then an attempt to enroll an agent with the old token fails
+
+@package-added-to-default-config
+Scenario: Adding packages to configurations
+  Given an agent is deployed to Fleet
+    And the package list API returns successfully
+  When the "Cisco" "latest" package version is installed successfully
+    And the "Cisco" package datasource is added to the "default" configuration
+  Then the "default" configuration shows the "Cisco" datasource added
+
+@new-agent-configuration
+Scenario: Assign an Agent to a new configuration
+  Given an agent is deployed to Fleet
+    And the agent is listed in Fleet as online
+  When a new configuration named "Test Fleet" is created
+    And the Agent is assigned to the configuration "Test Fleet"
+  Then a new enrollment token is created
+    And there is new data in the index from agent
```

Comment:
What about rephrasing this to:

```gherkin
Given a new configuration named "Test Fleet" is created
When an agent is deployed to Fleet
    And the Agent is assigned to the configuration "Test Fleet"
Then a new enrollment token is created
```

I'd remove the `there is new data...` step for the reasons described in the issue: we do not have more information about what to check

Reply:
I think we can remove this and re-add it when possible... I do however think that every few weeks I'm learning more about how to assess the data flowing into ES.  I think we can craft a better query now, adding in the source of the data (being either of Metricbeat, Filebeat, or Agent) to assert on this.  At a high level we don't care which is sending data, but a good short term goal would be to assert each of them has at least sent 1 document.  Agent may not send much more than that, unless there are problems.

Reply:
Ok, let's use this new scenario. We can create an issue to add the step for checking specific data. We can use it to fill in with your research

Reply:
Is it required that the agent is listed on fleet for this scenario?